### DISCUSSION TITLE
Help on component in `use server` and using form#action

### DISCUSSION THREAD

**Discussion by nkpgardose** [storybookjs/storybook#28317] [Help]
### Summary

```ts
'use server'
import SubmitButton from './SubmitButton'
import { signOutAction } from './actions'

function SignoutButton() {
	return (
		<form action={signOutAction}>
			<SubmitButton />
		</form>
	)
}

export default SignoutButton
```

This is a simple RSC component in NextJS. It uses server action, which will be a common approach in all react components in the future, and I've created a simple storybook

```ts
import type { Meta, StoryObj } from '@storybook/react'
import SignoutButton from './SignoutButton'

const meta: Meta<typeof SignoutButton> = {
	title: 'Components / SignoutButton',
	component: SignoutButton,
	decorators: [
		(Story) => (
			<div style={{ padding: 'var(--spacing-m)' }}>
				<Story />
			</div>
		),
	],
}

export default meta
type SignoutButtonStory = StoryObj<typeof SignoutButton>

export const Default: SignoutButtonStory = {
	args: {},
	parameters: {
		nextjs: {
			appDirectory: true,
		},
	},
}
```

I'm having an issue with the rendered `action` value here.

```
<form action="javascript:throw new Error('A React form was unexpectedly submitted. If you called form.submit() manually, consider using form.requestSubmit() instead. If you\'re trying to use event.stopPropagation() in a submit event handler, consider also calling event.preventDefault().')">...</form>
```

and ultimately, it may also cause some errors when rendering the component(since it might import backend-like package). In this case , inside `./actions`.

I tried this approach but it didn't work. https://storybook.js.org/docs/writing-stories/mocking-modules#using-mocked-modules-in-stories

Can anyone help here? Or what would be your solution in case you've encountered this issue?

**Comment by musjj** [comment]
Any reason why this was closed? Did you find a solution?
